### PR TITLE
Add bash runner script for Mask R-CNN on TPU

### PR DESCRIPTION
**Pull request by sprt**
... so that you guys can easily repro what I've been doing.

Also added a couple comments for history.

### CODE REVIEW

File: references/detection/run_mask_rcnn_tpu.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# TPU XLA
+export TPU_IP_ADDRESS="10.2.101.2"
+export XRT_TPU_CONFIG="tpu_worker;0;$TPU_IP_ADDRESS:8470"
+
+# CPU XLA
+#export XRT_DEVICE_MAP="CPU:0;/job:localservice/replica:0/task:0/device:XLA_CPU:0"
+#export XRT_WORKERS="localservice:0;grpc://localhost:40934"
+
+export TF_CPP_VMODULE=tensor=5 # logging
+export TRIM_GRAPH_SIZE=100000
```
Comment (jysohn23) [inline references/detection/run_mask_rcnn_tpu.sh]: Are these absolutely needed?

File: references/detection/run_mask_rcnn_tpu.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+# TPU XLA
+export TPU_IP_ADDRESS="10.2.101.2"
+export XRT_TPU_CONFIG="tpu_worker;0;$TPU_IP_ADDRESS:8470"
+
+# CPU XLA
+#export XRT_DEVICE_MAP="CPU:0;/job:localservice/replica:0/task:0/device:XLA_CPU:0"
+#export XRT_WORKERS="localservice:0;grpc://localhost:40934"
+
+export TF_CPP_VMODULE=tensor=5 # logging
```
Comment (jysohn23) [inline references/detection/run_mask_rcnn_tpu.sh]: Let's remove this VLOG for just runner.

File: references/detection/run_mask_rcnn_tpu.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+export TRIM_GRAPH_SIZE=100000
+export TRIM_GRAPH_CHECK_FREQUENCY=1000
+
+export XLA_DENSE_SCATTER_FACTOR=100
```
Comment (jysohn23) [inline references/detection/run_mask_rcnn_tpu.sh]: What does this do?

File: torchvision/ops/boxes.py [hunk 4]
```diff
@@ -30,6 +30,8 @@ def nms(boxes, scores, iou_threshold, post_nms_top_n):
         by NMS, sorted in decreasing order of scores
     """
     device = boxes.device
+    # XXX: adding this mark_step() call speeds up Mask R-CNN inference on TPU
+    # from 0.9 sec/im to 0.55 sec/im, reason unsure
```
Comment (jysohn23) [inline torchvision/ops/boxes.py:34]: Can you just add a "TODO" for easier searching later 😄

File: references/detection/run_mask_rcnn_tpu.sh (hunk 1 above)
Comment (sprt) [reply references/detection/run_mask_rcnn_tpu.sh]: Not `TF_CPP_VMODULE` of course. :smiley:
`TRIM_GRAPH_SIZE` is probably not needed either, I added it back when I was running into a lot of XLA issues and I think I either never looked at it or set it to the default value. Let me check what's happening without it.

File: references/detection/run_mask_rcnn_tpu.sh (hunk 2 above)
Comment (sprt) [reply references/detection/run_mask_rcnn_tpu.sh]: Will do.

File: references/detection/run_mask_rcnn_tpu.sh (hunk 3 above)
Comment (sprt) [reply references/detection/run_mask_rcnn_tpu.sh]: Magic flag suggested by Davide when I was running into scatter issues and OOMing, will check behavior without that flag.

File: torchvision/ops/boxes.py (hunk 4 above)
Comment (sprt) [reply torchvision/ops/boxes.py:34]: Done

File: references/detection/run_mask_rcnn_tpu.sh (hunk 3 above)
Comment (sprt) [reply references/detection/run_mask_rcnn_tpu.sh]: Got rid of all these flags and no issue at all!

**Review by jysohn23 — APPROVED** [review]

**Review by taylanbil — APPROVED** [review]

**Pull request merged**